src/index.js

```diff
@@ -1,32 +1,67 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const API = 'https://rickandmortyapi.com/api/character/';
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const fetchData = url_api => {
+  return new Promise((resolve, reject) => {
+    const xhttp = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+    xhttp.onreadystatechange = event => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) resolve(xhttp.responseText)
```

Comment:
Para mantener un código limpio, te recomiendo hacer el parse de "JSON.parse()" en el resolve de tu función y no hacer JSON.parse por cada llamado. 

```suggestion
        resolve(JSON.parse(xhttp.responseText));
```